IntroOOP/BankAccount/BankAccount.cs

```diff
@@ -150,4 +144,34 @@ public bool TransferMoney(BankAccount source, decimal amount)
     /// Метод печатает состояние
     /// </summary>
     public void GetInfo() { Console.WriteLine("Номер счета: {0}\nБаланс: {1:N} RUB\nТип: {2}\n", ID, Balance, Type); }
+
+
+    public static bool operator ==(BankAccount a, BankAccount b) => a.Equals(b);
+    public static bool operator !=(BankAccount a, BankAccount b) => !(a == b);
+
+    public override bool Equals(object? obj)
+    {
+        if (obj == null) return false;
+        if(obj.GetType() != typeof(BankAccount)) return false;
+
+        var other = (BankAccount)obj;
+
+        return Balance == other.Balance && Type == other.Type;
```

Comment:
```suggestion
        if (obj is not BankAccount other) 
            return false;

        return _ID == other._ID;
                // && Balance == other.Balance  // если Id уникален, то остальное можно не проверять.
                // && Type == other.Type;
```
Можно воспользоваться "сопоставлением с образцом"